### PR TITLE
Allow https metadata without credentials

### PR DESCRIPTION
**Pull request by Gabee01**
[Issue 56](https://github.com/OData/lab/issues/56)

### CODE REVIEW

File: ODataConnectedService/src/ViewModels/ConfigODataEndpointViewModel.cs [hunk 1]
```diff
@@ -77,6 +77,8 @@ private string GetMetadata(out Version edmxVersion)
                 }
             }
 
+            ServicePointManager.ServerCertificateValidationCallback += new System.Net.Security.RemoteCertificateValidationCallback((s, ce, ch, ssl) => true);
```
Comment (AlanWong-MS) [inline ODataConnectedService/src/ViewModels/ConfigODataEndpointViewModel.cs:80]: Is this meant for test/development environment only, or have you encountered issues outside of the development process?

In either case, we should add some guards inside the lambda. We should add a boolean such as isDevelopment which returns true as presented in your code. We should also add validation when isDevelopment == false. I would also recommend using more descriptive variable names. Below is my comprehensive suggestion:

```
ServicePointManager.ServerCertificateValidationCallback +=
    new System.Net.Security.RemoteCertificateValidationCallback((sender, certificate, chain, sslPolicyErrors) =>
    {
        if (isDevelopment)
        {
            return true;
        }

        if (sslPolicyErrors == System.Net.Security.SslPolicyErrors.None)
        {
            return true;
        }

        // Do not allow this client to communicate with unauthenticated servers.
        return false;
    });
```

The `isDevelopment` would be a represented as a checkbox in the UI that you can add in `ODataConnectedService/src/Views/AdvancedSettings.xaml`. You would have to bind that checkbox with the boolean.

**Review by AlanWong-MS — CHANGES_REQUESTED** [review]
Great start. Let's add some validations to it.

**Comment by nla-brandonjames** [comment]
He has deleted this repository

**Comment by AlanWong-MS** [comment]
Closing this PR due to contributor's repo deletion.

**Comment by onyangofred** [comment]
Just open the link in Internet explorer, will prompt for username and password. Then save the credentials. Try again in visual studio. It works!!!

**Pull request closed without merging**